packages/shared/src/features/common/components/FormInputCheckboxGroup.tsx

```diff
@@ -0,0 +1,151 @@
+import type { ComponentProps, PropsWithChildren, ReactElement } from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import {
+  Button,
+  ButtonVariant,
+  ButtonSize,
+} from '../../../components/buttons/Button';
+import type { ButtonProps } from '../../../components/buttons/Button';
+
+type CheckboxValue = string;
+type CheckboxValues = string[];
+
+export interface CheckboxItem {
+  image?: ComponentProps<'img'>;
+  label: string;
+  value: CheckboxValue;
+}
+
+export enum CheckboxGroupVariant {
+  Horizontal = 'horizontal',
+  Vertical = 'vertical',
+}
+
+export interface FormInputCheckboxGroupProps
+  extends PropsWithChildren<ComponentProps<'div'>> {
+  /**
+   * @default []
+   */
+  defaultValue?: CheckboxValues;
+  name: string;
+  onValueChange?: (value: CheckboxValues) => void;
+  /**
+   * @default []
+   */
+  options?: CheckboxItem[];
+  /**
+   * @default 1
+   */
+  cols?: number;
+  value?: CheckboxValues;
+  /**
+   * @default CheckboxGroupVariant.Horizontal
+   */
+  variant?: CheckboxGroupVariant;
+}
+
+const getChangedValue = (
+  value: CheckboxValues,
+  selectedValue: CheckboxValue,
+): CheckboxValues => {
+  const isSelected = value.includes(selectedValue);
+  return isSelected
+    ? value.filter((item) => item !== selectedValue)
+    : [...value, selectedValue];
+};
+
+const FormInputCheckbox = ({
```

Comment:
More of a question, don't we want a option here where it can only have 1 option (radio basically)

Reply:
Yes, but since we need to move to the next step on the first selection, user won't be able to pick a second one.
 
I was thinking of placing this logic in the quiz step to avoid adding more complexity to this component.  Maybe when we reuse it in the future, we could add this behavior. Wdyt?

Reply:
Hmm ok so you can't chance selection?
Sorry wasn't visible from the storybook.

I guess it's fine we can tackle once we need it?

Reply:
You can go back and change your choice, but when clicking on option you will progress to the next one. 
Yes, we can add this behaviour easily 💯